### PR TITLE
withAzureEnv step

### PR DESCRIPTION
**Pull request by v1v**
## What does this PR do?

withAzureEnv step to be consumed by 

## Why is it important?

Simplify the pipeline with some logic in the shared library

## Related issues

Required by https://github.com/elastic/azure-vm-extension/pull/13e

### CODE REVIEW

File: vars/withAzureEnv.groovy [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    [var: 'AZ_PASSWORD', password: password],
+    [var: 'AZ_TENANT', password: tenant],
+    [var: 'AZ_SUBSCRIPTION', password: subscription]
+  ]){
+    body()
+  }
+}
+
+def validateField(String value, String errorMessage) {
+  if(value == null){
```
Comment (mdelapenya) [inline vars/withAzureEnv.groovy:52]: null or empty?

Comment (v1v) [reply vars/withAzureEnv.groovy:52]: both cases, so I'll include that validation too :)

Comment (v1v) [reply vars/withAzureEnv.groovy:52]:
```suggestion
  if(value == null || value?.trim() == ""){
```

**Review by mdelapenya — APPROVED** [review]

**Review by cachedout — APPROVED** [review]

File: src/test/groovy/ApmBasePipelineTest.groovy [hunk 2]
```diff
@@ -596,6 +597,8 @@ class ApmBasePipelineTest extends DeclarativePipelineTest {
     if(VaultSecret.SECRET_AZURE.equals(s)){
       return [data: [ client_id: 'client_id_1', client_secret: 'client_secret_1', subscription_id: 'subscription_id_1', tenant_id: 'tenant_id_1' ]]
     }
+    if(VaultSecret.SECRET_AZURE_VM_EXTENSION.equals(s)){
+      return [data: [ password: 'password_1', username: 'username_1', subscription: 'subscription_id_1', tenant: 'tenant_id_1' ]]
```
Comment (v1v) [inline src/test/groovy/ApmBasePipelineTest.groovy:601]:
```suggestion
    if(VaultSecret.SECRET_AZURE_VM_EXTENSION.equals(s)){
      return [data: [ password: 'password_1', username: 'username_1', subscription: 'subscription_id_1', tenant: 'tenant_id_1' ]]
    }
```

**Pull request merged**